hw_lesson13.py

```diff
@@ -0,0 +1,107 @@
+import inspect
+from functools import wraps
+
+"""
+Task 1
+Write a decorator that prints a function with arguments passed to it.
+NOTE! It should print the function, not the result of its execution!
+"""
+
+
+def logger(func):
+    def wrap():
+        code_func = inspect.getsource(func)
+
+        split_for_return = code_func.split('return')
+
+        give_code = split_for_return[-1]
+
+        return give_code
+
+    return print(wrap())
+
+
+@logger
+def add(x, y):
+    return x + y
+
+
+@logger
+def square_all(*args):
+    return [arg ** 2 for arg in args]
+
+
+"""
+Task 2
+
+Write a decorator that takes a list of stop words and replaces them with * inside the decorated function
+"""
+
+
+def stop_words(words: list):
+    def wrap(func):
+        @wraps(func)
+        def wrapper(*args, **kwargs):
+            string: str = func(*args, **kwargs)
+
+            for i in range(len(words)):
```

Comment:
Actualy, we can do just:
```suggestion
            for word in words:
                string = string.replace(word, '*')
```

Reply:
I have error "list indices must be integers or slices, not str" if i changed for word in words:
and i havn't error when use ' for i in range(len(words)):'